photon-core/src/main/java/org/photonvision/vision/pipe/impl/FindBoardCornersPipe.java

```diff
@@ -219,16 +232,92 @@ private Size getWindowSize(MatOfPoint2f inPoints) {
     private FindBoardCornersPipeResult findBoardCorners(Pair<Mat, Mat> in) {
         createObjectPoints();
 
+        float[] levels = null;
+        var outLevels = new MatOfFloat();
+
+        var objPts = new MatOfPoint3f();
+        var outBoardCorners = new MatOfPoint2f();
+
         var inFrame = in.getLeft();
         var outFrame = in.getRight();
 
         // Convert the inFrame too grayscale to increase contrast
         Imgproc.cvtColor(inFrame, inFrame, Imgproc.COLOR_BGR2GRAY);
         boolean boardFound = false;
 
-        if (params.type == UICalibrationData.BoardType.CHESSBOARD) {
+        // Get the size of the inFrame
+        this.imageSize = new Size(inFrame.width(), inFrame.height());
+
+        if (params.type == UICalibrationData.BoardType.CHARUCOBOARD) {
+            Mat ObjPoints =
+                    new Mat(); // 3 dimensional currentObjectPoints, the physical target ChArUco Board
+            Mat imgPoints =
+                    new Mat(); // 2 dimensional currentImagePoints, the likely distorted board on the flat
+            // camera sensor frame posed relative to the target
+            Mat detectedCorners = new Mat(); // currentCharucoCorners
+            Mat detectedIds = new Mat(); // currentCharucoIds
+            detector.detectBoard(inFrame, detectedCorners, detectedIds);
+
+            // reformat the Mat to a List<Mat> for matchImagePoints
+            final List<Mat> detectedCornersList = new ArrayList<>();
+            for (int i = 0; i < detectedCorners.total(); i++) {
+                detectedCornersList.add(detectedCorners.row(i));
+            }
+            if (detectedCornersList.size() > 0) {
+                boardFound = true;
+            }
+
+            if (!boardFound) {
+                // If we can't find a board, give up
+                return null;
+            }
+            board.matchImagePoints(detectedCornersList, detectedIds, ObjPoints, imgPoints);
+
+            // draw the charuco board
+            Objdetect.drawDetectedCornersCharuco(
+                    outFrame, detectedCorners, detectedIds, new Scalar(0, 0, 255)); // Red Text
+
+            if (!boardFound) {
+                // If we can't find a charuco board, give up
+                return null;
+            }
+
+            imgPoints.copyTo(outBoardCorners);
+            ObjPoints.copyTo(objPts);
+
+            if (params.useMrCal) {
```

Comment:
Why does this work only need to happen for mrcal? Again document

Reply:
this still seems smelly. Why should the choice of backend change the calibration detection result?